model/HasManyList.php

```diff
@@ -45,7 +45,7 @@ protected function foreignIDFilter($id = null) {
 		if ($id === null) $id = $this->getForeignID();
 
 		// Apply relation filter
-		$key = "\"$this->foreignKey\"";
+		$key = "\"$this->dataClass\".\"$this->foreignKey\"";
```

Comment:
You need to check the table the field is actually on; E.g. ParentID is on SiteTree table, not on ErrorPage table. Can you please use the ClassInfo::table_for_object_field() API?

A unit test to demonstrate this would help too please. :)